examples/library/test_manycore_vcache_sequence.c

```diff
@@ -56,16 +56,17 @@ int test_manycore_vcache_sequence() {
         }
 
         const hb_mc_config_t *config = hb_mc_manycore_get_config(mc);
-        uint32_t manycore_dim_x = hb_mc_coordinate_get_x(hb_mc_config_get_dimension_vcore(config));
-
-        uint32_t dram_coord_x = 0;
-        uint32_t dram_coord_y = hb_mc_config_get_dram_y(config);
         int mismatch = 0;
 
         /**************************************************************/
         /* Loop over all DRAM banks and write ARRAY_LEN words to each */
         /**************************************************************/
-        for (dram_coord_x = 0; dram_coord_x < 1; dram_coord_x++) {
+        hb_mc_coordinate_t dram_coord;
+        hb_mc_config_foreach_dram_coordinate(dram_coord, config)
+        {
```

Comment:
This is pretty excellent.

Reply:
Yeah... I remembered having to update these tests when we added vcaches to the north of the array. I figured we could do this and it might spare us the pain going forward. There is some code bsg_manycore.cpp that could potentially make use of these, too.

Reply:
Macros like this are pretty common in C code because they hide loop logic that distracts from the logic of the program. It should be noted that modern C++ does the same thing (and clearly better IMO) with this syntax:
```c++
for (type iter : container) { }
```

Reply:
OTHO the lines of code required to write a complex random access iterator might be more verbose than this macro so...

Reply:
Do what you feel is best

Reply:
Well there's no choice here: We're a C API.